marketplace/mux-video-uploader/src/index.tsx

```diff
@@ -101,6 +113,53 @@ export class App extends React.Component<AppProps, AppState> {
     return headers;
   };
 
+  requestDeleteAsset = async () => {
+    if (!this.state.value || !this.state.value.assetId) {
+      throw Error(
+        'Something went wrong, we cannot delete an asset without an assetId.'
+      );
+    }
+
+    const result = await this.props.sdk.dialogs.openConfirm({
+      title: "Are you sure you want to delete this asset?",
+      message: "This will remove the asset in Mux and in Contentful. There is no way to recover your video, make sure you have a backup if you think you may want to use it again.",
+      intent: "negative",
+      confirmLabel: "Yes, delete this asset",
+      cancelLabel: "Nevermind",
+    });
+
+    if (!result) {
+      this.setState({ isDeleting: false });
+      return;
+    }
+    this.setState({ isDeleting: true });
+
+    const res = await fetch(`https://api.mux.com/video/v1/assets/${this.state.value.assetId}`, {
+      ...this.muxBaseReqOptions,
+      method: 'DELETE',
+    });
+
+    if (res.status === 401) {
+      throw Error(
+        'Looks like something is wrong with the Mux Access Token specified in the config. Are you sure the token ID and secret in the extension settings match the access token you created?'
+      );
+    }
+
+    await this.resetField();
+    this.setState({ isDeleting: false });
+  };
+
+  resetField = async () => {
+    await this.props.sdk.field.setValue({
+      uploadId: undefined,
+      assetId: undefined,
+      playbackId: undefined,
+      ready: undefined,
+      ratio: undefined,
+    });
```

Comment:
Is there a cleaner way to unset all the fields of an object?

Reply:
There is `field.removeValue` method however, that would set the value to `undefined`. I think what you have here is OK because the functionality the SDK provides wouldn't reset your custom JSON.